packages/framework/ai-collab/README.md

```diff
@@ -1,34 +1,194 @@
+const userAsk = "Update the task group description to be a about creating a new Todo list application. Create a set of tasks to accomplish this and assign them to the available engineers. Keep in mind the max capacity of each engineer as you assign tasks."
+
+// Collaborate with AI one function call.
+const response = await aiCollab<typeof PlannerAppState>({
+		openAI: {
+			client: new OpenAI({
+				apiKey: OPENAI_API_KEY,
+			}),
+			modelName: "gpt-4o",
+		},
+		treeView: view,
+		treeNode: view.root.taskGroups[0],
+		prompt: {
+			systemRoleContext:
+				"You are a manager that is helping out with a project management tool. You have been asked to edit a group of tasks.",
+			userAsk: userAsk,
+		},
+		planningStep: true,
+		finalReviewStep: true,
+		dumpDebugLog: true,
+	});
+
+if (response.status === 'sucess') {
+	// Render the UI view of your task groups.
+	window.alert(`The AI has successfully completed your request.`);
+} else {
+	window.alert(`Something went wrong! response status: ${response.status}, error message: ${response.errorMessage}`);
+}
+
+
+```
+
+Once the `aiCollab` function call is initiated, an LLM will immediately begin attempting to make changes to your Shared Tree using the provided user prompt, the types of your SharedTree and the provided app guidance. The LLM produces multiple changes, in a loop asynchronously. Meaning, you will immediatley see changes if your UI's render loop is connected to your SharedTree App State.
+
+### Example 2: Collaborate with AI onto a branched state and let the user merge the review and merge the branch back manually
+- **Coming Soon**
+
+
+## Folder Structure
+
+- `/explicit-strategy`: The new explicit strategy, utilizing the prototype built during the fall FHL, with a few adjustments.
+     - `agentEditReducer`: This file houses the logic for taking in a `TreeEdit`, which the LLM produces, and applying said edit to the
+     -  actual SharedTree.
+     - `agentEditTypes.ts`: The types of edits an LLM is prompted to produce in order to modify a SharedTree.
+     - `idGenerator.ts`: `A manager for producing and mapping simple id's in place of UUID hashes when generating prompts for an LLM
+     - `jsonTypes.ts`: utility JSON related types used in parsing LLM response and generating LLM prompts.
+     - `promptGeneration.ts`: Logic for producing the different types of prompts sent to an LLM in order to edit a SharedTree.
+     - `typeGeneration.ts`: Generates serialized(/able) representations of a SharedTree Schema which is used within prompts and the generated of the structured output JSON schema
+     - `utils.ts`: Utilities for interacting with a SharedTree
+- `/implicit-strategy`: The original implicit strategy, currently not used under the exported aiCollab API surface.
+
+## Known Issues & limitations
+1. Union types for a TreeNode are not present when generating App Schema. This will require extracting a field schema instead of TreeNodeSchema when passed a non root node.
```

Comment:
```suggestion
## Known Issues & limitations

1. Union types for a TreeNode are not present when generating App Schema. This will require extracting a field schema instead of TreeNodeSchema when passed a non root node.
```

Reply:
Updated in commit b700ced170a403db24c887b25ce4917392948b3c